packages/twenty-front/src/modules/views/states/contexts/CurrentViewComponentInstanceContext.tsx

```diff
@@ -0,0 +1,28 @@
+import { useRecoilComponentValueV2 } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentValueV2';
+import { createComponentInstanceContext } from '@/ui/utilities/state/component-state/utils/createComponentInstanceContext';
+import { currentViewIdComponentState } from '@/views/states/currentViewIdComponentState';
+
+export const CurrentViewComponentInstanceContext =
+  createComponentInstanceContext();
+
+type CurrentViewComponentInstanceContextProviderProps = {
+  recordIndexId: string;
+  children: React.ReactNode;
+};
+
+export const CurrentViewComponentInstanceContextProvider = ({
+  recordIndexId,
+  children,
+}: CurrentViewComponentInstanceContextProviderProps) => {
+  const currentViewId =
+    useRecoilComponentValueV2(currentViewIdComponentState, recordIndexId) ??
+    'none';
+
+  return (
+    <CurrentViewComponentInstanceContext.Provider
```

Comment:
Do we need this ? There's already `ViewComponentInstanceContext`

Reply:
Yes the difference between `ViewComponentInstanceContext` and `CurrentViewComponentInstanceContext` is that `CurrentViewComponentInstanceContext` is actually storing the current view id and `ViewComponentInstanceContext` is just storing the plural record name as scope.
Maybe `ViewComponentInstanceContext` is not correctly named.

Reply:
I deleted the additional `CurrentViewComponentInstanceContext` because it was indeed redundant as we can access our viewId with the already existing hook. The solution was to use `ViewComponentInstanceContext` on the new component state.

Reply:
@lucasbordeau I'm still not sure about that as the `scopeId` stored inside `ViewComponentInstanceContext` is the object plural name, and view-groups are different based on the view.
So if we change the view, we can have a glitch with the previous view-groups until the new ones load if we are just basing the state on the object plural name.